test/common/event/dispatched_thread_impl_test.cc

```diff
@@ -22,15 +23,17 @@ namespace Event {
 class DispatchedThreadTest : public testing::Test {
 protected:
   DispatchedThreadTest()
-      : config_(1000, 1000, 1000, 1000), guard_dog_(fakestats_, config_, test_time_.timeSystem()),
-        thread_(test_time_.timeSystem()) {}
+      : config_(1000, 1000, 1000, 1000), api_(std::chrono::milliseconds(1000)),
```

Comment:
Can we add a default constructor for `Api::Impl`? It's a bit hard to understand the significance of this parameter in various tests.